### PR TITLE
doc(PR template): Make automated tests an explicit item in the PR template.

### PR DESCRIPTION
**Pull request by petervdmeer**
**Feature:** This PR extends the template with a section regarding automated testing, initially discussed in #4308

## Feature Details
Make automated (regression) tests an explicit entry in the PR template.

## UI Screenshots / Usage Examples
N/A. Not visible in-game. And just fill the template with all required details.

## Testing Done
Used the new template header here in the PR.

### Automated tests added
Extended the template with a section explicitly for this.

## Performance Impact
N/A. But considering automated tests earlier in the PR process might help to get things merged more efficiently.

### CODE REVIEW

**Comment by Amazinite** [comment]
It's to my understanding that at the moment not everything can have automated tests made for them. Perhaps instead of having a blanket question of whether the poster has created automated tests for their code change, we describe a list of code changes that should have automated tests.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -46,5 +46,8 @@ This save file can be used to verify the bugfix. The bug will occur when using {
 ## Testing Done
 {{describe how you tested the new feature}}
 
+### Automated tests added
```
Comment (Amazinite) [inline .github/pull_request_template.md:49]: All the other headers are capitalized.
```suggestion
### Automated Tests Added
```

**Comment by petervdmeer** [comment]
> It's to my understanding that at the moment not everything can have automated tests made for them. Perhaps instead of having a blanket question of whether the poster has created automated tests for their code change, we describe a list of code changes that should have automated tests. Similar to how the performance impact sections says you can put "N/A" if you haven't touched performance-critical code.

Let's aim for having all of the features in the game regression tested. The first step to get there is to change the policy from **"we don't do automatic tests unless we are sure we can do it"** to **"we do automatic tests unless we are sure we cannot do it"**.

The "Testing Done" section in [your recent PR](https://github.com/endless-sky/endless-sky/pull/7210) is a great example of tests that could have been automated. Loading savegames and checking conditions is something which the integration testing framework supports very well, and the [on-demand autoconditions](https://github.com/endless-sky/endless-sky/pull/5244) were originally developed to make the integration testing framework stronger. Once people find out how to work with nested prefixed providers under a special `"test: "` prefixed provider, then we will be able to expose really a lot of internal game state towards the integration testing framework.

[Splitting off some classes](https://github.com/endless-sky/endless-sky/pull/7128) should also improve testability using the Catch unit-test framework and [switching to a single build-system](https://github.com/endless-sky/endless-sky/pull/7020) should just improve any type of development (including test-development).

We will never be able to test everything, so some exceptions will always exist. But the list of exceptions should be tiny.

**Comment by petervdmeer** [comment]
Let's just do this. (We can always change the template later if we want to reword it or change it.)

**Pull request merged**